packages/material-ui-based-components/src/components/Checkbox.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useCallback } from 'react';
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import {
+  Checkbox,
+  FormControl, FormControlLabel, FormGroup, FormHelperText, Switch
+} from '@mui/material';
+
+const CheckboxComponent = (props: PROPS) => {
+  const {
+    label, id, required, enabled, value, errorMessage,
+    description, isError, properties, name
+  } = props;
+  const selectedValue = props.enum?.[0];
+  const unselectedValue = (props.enum?.length || 0) < 2 ? null : props.enum?.[1];
+  const k = 'afs:layout';
+  let row = false;
+  if (properties && properties[k]) {
+    row = properties[k].orientation === 'horizontal';
+  }
+
+  const control = () => {
+    if (props[':type'] === 'switch') {
+      return (<Switch />);
+    }
+    else {
+      return (<Checkbox />);
+    }
+  };
+
+  const changeHandler = useCallback((event: any) => {
+    const val = (event.target.checked) ? selectedValue : unselectedValue;
+    props.dispatchChange(val);
+  }, [props.dispatchChange]);
+
+  return (
+    <FormControl
+      variant="standard"
```

Comment:
variant should be dynamic. It will come from the props.properties['afs:layout'].variant

Reply:
Added dynamic variant